packages/ember-htmlbars/tests/helpers/view_test.js

```diff
@@ -366,436 +61,99 @@ QUnit.test('Should apply class without condition always', function() {
-
+QUnit.test('Should not apply classes when bound property specified is false', function() {
   view = EmberView.create({
-    template: compile(template)
+    controller: {
+      someProp: false
+    },
+    template: compile('{{#view id="foo" class=someProp}} Foo{{/view}}')
   });
 
   runAppend(view);
 
-  var childView = firstGrandchild(view);
-  equal(childView, get(firstChild(childView), 'parentView'), 'parent view is correct');
+  ok(!jQuery('#foo').hasClass('some-prop'), 'does not add class when value is falsey');
 });
 
-QUnit.test('Child views created using the view helper should have their IDs registered for events', function() {
-  var template = '{{view}}{{view id="templateViewTest"}}';
-
+QUnit.test('Should apply classes of the dasherized property name when bound property specified is true', function() {
   view = EmberView.create({
-    template: compile(template)
+    controller: {
+      someProp: true
+    },
+    template: compile('{{#view id="foo" class=someProp}} Foo{{/view}}')
   });
 
   runAppend(view);
 
-  var childView = firstChild(view);
-  var id = childView.$()[0].id;
-  equal(EmberView.views[id], childView, 'childView without passed ID is registered with View.views so that it can properly receive events from EventDispatcher');
-
-  childView = nthChild(view, 1);
-  id = childView.$()[0].id;
-  equal(id, 'templateViewTest', 'precond -- id of childView should be set correctly');
-  equal(EmberView.views[id], childView, 'childView with passed ID is registered with View.views so that it can properly receive events from EventDispatcher');
+  ok(jQuery('#foo').hasClass('some-prop'), 'adds dasherized class when value is true');
 });
 
-QUnit.test('Child views created using the view helper and that have a viewName should be registered as properties on their parentView', function() {
-  var template = '{{#view}}{{view viewName="ohai"}}{{/view}}';
-
+QUnit.test('should escape HTML in normal mustaches', function() {
   view = EmberView.create({
-    template: compile(template)
+    template: compile('{{view.output}}'),
+    output: 'you need to be more <b>bold</b>'
   });
 
   runAppend(view);
+  equal(view.$('b').length, 0, 'does not create an element');
+  equal(view.$().text(), 'you need to be more <b>bold</b>', 'inserts entities, not elements');
 
-  var parentView = firstChild(view);
-  var childView  = firstGrandchild(view);
+  run(function() {
+    set(view, 'output', 'you are so <i>super</i>');
+  });
 
-  equal(get(parentView, 'ohai'), childView);
+  equal(view.$().text(), 'you are so <i>super</i>', 'updates with entities, not elements');
+  equal(view.$('i').length, 0, 'does not create an element when value is updated');
 });
 
-QUnit.test('{{view}} id attribute should set id on layer', function() {
-  owner.register('template:foo', compile('{{#view view.idView id="bar"}}baz{{/view}}'));
-
-  var IdView = EmberView;
-
+QUnit.test('should not escape HTML in triple mustaches', function() {
```

Comment:
Now covered in https://github.com/emberjs/ember.js/pull/13236/files#diff-c47395a5aaf92aa0a7edbfa812d4e0afR687